pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -185,70 +192,38 @@ func (op *ObjectPatch) Apply(c client.Client, labels map[string]string) error {
 		return nil
 	}
 
-	return c.Patch(context.TODO(), op.obj, op.patch)
+	return c.Patch(ctx, op.obj, op.patch)
 }
 
 // Reconcile reconciles the driver deployment. When adding new
 // objects, extend also currentObjects above and the RBAC rules in
 // deploy/kustomize/operator/operator.yaml.
-func (d *PmemCSIDriver) Reconcile(r *ReconcileDeployment) error {
+func (d *pmemCSIDeployment) reconcile(ctx context.Context, r *ReconcileDeployment) error {
 
 	if err := d.EnsureDefaults(r.containerImage); err != nil {
 		return err
 	}
+	l := logger.Get(ctx).WithName("reconcile")
 
-	klog.Infof("Deployment: %q, state %q ", d.Name, d.Status.Phase)
+	l.V(3).Info("start", "deployment", d.Name, "phase", d.Status.Phase)
 	var allObjects []apiruntime.Object
 	redeployAll := func() error {
 		var o apiruntime.Object
 		var err error
-		s, err := d.redeploySecrets(r)
+		s, err := d.redeploySecrets(ctx, r)
 		if err != nil {
 			return err
 		}
 		for _, o := range s {
 			allObjects = append(allObjects, o)
 		}
-		if o, err = d.redeployProvisionerRole(r); err != nil {
-			return fmt.Errorf("failed to update RBAC role: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployProvisionerRoleBinding(r); err != nil {
-			return fmt.Errorf("failed to update RBAC role bindings: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployProvisionerClusterRole(r); err != nil {
-			return fmt.Errorf("failed to update RBAC cluster role: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployProvisionerClusterRoleBinding(r); err != nil {
-			return fmt.Errorf("failed to update RBAC cluster role bindings: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployServiceAccount(r); err != nil {
-			return fmt.Errorf("failed to update service account: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployControllerService(r); err != nil {
-			return fmt.Errorf("failed to update controller service: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployMetricsService(r); err != nil {
-			return fmt.Errorf("failed to update controller service: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployCSIDriver(r); err != nil {
-			return fmt.Errorf("failed to update CSI driver: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployControllerDriver(r); err != nil {
-			return fmt.Errorf("failed to update controller driver: %v", err)
-		}
-		allObjects = append(allObjects, o)
-		if o, err = d.redeployNodeDriver(r); err != nil {
-			return fmt.Errorf("failed to update node driver: %v", err)
+
+		for name, handler := range subObjectHandlers {
```

Comment:
These changes are not related logging, would be nice to keep them a separate commit/PR.

Reply:
While it would be nicer, that implies doing the code cleanup work twice because the change was done in parallel to adding ctx (basically I noticed that I would have to make the same change multiple times and instead decided to reduce the amount of similar functions) and now does not apply to "devel".

I don't think the "nicer" part is worth the extra effort.